File: F1WebAppFront/F1WebApp/src/types/forum.ts
export interface ForumComment {
    id: number;
    comment: string;
    createDate: string;
    userId: number;
    username: string;
}

export interface ForumPost {
    id: number;
    title: string;
    description: string;
    userId: number;
    username: string;
    comments: ForumComment[];
}

File: F1WebAppBack/F1.Shared.Domain/Comunity/Entities/Forum.cs
using F1.Shared.Domain.Comunity.Entities.Interfaces;
using F1.Shared.Domain.Users.Entities.Interfaces;

namespace F1.Shared.Domain.Comunity.Entities
{
    public class Forum : IForum
    {
        public long Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public IUser User { get; set; } = null!;
        public DateTime? CreateDate { get; set; }
        public IEnumerable<IForumComment> Comments { get; set; } = [];
    }
}
